driver/normalizer/normalizer.go

```diff
@@ -28,31 +28,53 @@ var Normalize = Transformers([][]Transformer{
 
 func funcDefMap(typ string, async bool) Mapping {
 	return MapSemantic(typ, uast.FunctionGroup{}, MapObj(
-		Obj{
-			"body": Var("body"),
-			"name": Var("name"),
+		Fields{
+			{Name: "body", Op: Var("body")},
+			{Name: "name", Op: Var("name")},
 			// Arguments should be converted by the uast.Arguments normalization
-			"args": Obj{
+			{Name: "args", Op: Obj{
 				"args":       Var("arguments"),
 				uast.KeyPos:  Var("_pos"),
 				uast.KeyType: Var("_type"),
+			}},
+			// Will be filled only if there is a Python3 type annotation
+			{Name: "returns", Optional: "ret_opt", Op: Cases("ret_case",
+				Is(nil),
+				Obj{
+					uast.KeyType:  String("BoxedName"),
+					"boxed_value": Var("ret_type"),
+					// No problem dropping this one, it's used by an internal interpreter optimization/cache
+					// without semantic meaning
+					"ctx": Any(),
+				}),
 			},
+			{Name: "decorator_list", Op: Var("func_decorators")},
 		},
 		Obj{
 			"Nodes": Arr(
 				Obj{
 					// FIXME: generator=true if it uses yield anywhere in the body
 					"async": Bool(async),
 				},
+				Obj{
+					"decorators": Var("func_decorators"),
```

Comment:
Can be added to the object above.

Reply:
Done in next commit (plus function comments in the same Obj).